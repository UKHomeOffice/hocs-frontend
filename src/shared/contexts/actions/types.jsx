const types = {
    CANCEL: 'CANCEL',
    REDIRECT: 'REDIRECT',
    REDIRECTED: 'REDIRECTED',
    SET_ERROR: 'SET_ERROR',
    UNSET_ERROR: 'UNSET_ERROR',
    UNSET_FORM: 'UNSET_FORM',
    UPDATE_CASE_SUMMARY: 'UPDATE_CASE_SUMMARY',
    UPDATE_API_STATUS: 'UPDATE_API_STATUS',
    CLEAR_API_STATUS: 'CLEAR_API_STATUS',
    UPDATE_PAGE_META: 'UPDATE_PAGE_META',
    UPDATE_FORM: 'UPDATE_FORM',
    UPDATE_FORM_DATA: 'UPDATE_FORM_DATA',
    UPDATE_FORM_ERRORS: 'UPDATE_FORM_ERRORS',
    UPDATE_LOCATION: 'UPDATE_LOCATION',
    UPDATE_DASHBOARD: 'UPDATE_DASHBOARD',
    CLEAR_DASHBOARD: 'CLEAR_DASHBOARD',
    UPDATE_WORKSTACK: 'UPDATE_WORKSTACK',
    CLEAR_WORKSTACK: 'CLEAR_WORKSTACK',
    UNSET_CASE_SUMMARY: 'UNSET_CASE_SUMMARY',
    UNSET_CORRESPONDENTS: 'UNSET_CORRESPONDENTS',
    UNSET_CASE_NOTES: 'UNSET_CASE_NOTES',
    PASS_FORWARD_PROPS: 'PASS_FORWARD_PROPS',
    UNSET_DOCUMENTS: 'UNSET_DOCUMENTS',
    UPDATE_CASE_ACTION_DATA: 'UPDATE_CASE_ACTION_DATA',
    UNSET_CASE_ACTION_DATA: 'UNSET_CASE_ACTION_DATA',
    UPDATE_CASE_DATA: 'UPDATE_CASE_DATA',
    UNSET_CASE_DATA: 'UNSET_CASE_DATA',
    SET_SCHEMA_TAB_ERROR: 'SET_SCHEMA_TAB_ERROR',
    UNSET_SCHEMA_TAB_ERROR: 'UNSET_SCHEMA_TAB_ERROR'
};

export default types;
