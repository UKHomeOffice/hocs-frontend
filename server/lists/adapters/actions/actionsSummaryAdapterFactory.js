const appealAdaptor = require('./appealsSummaryAdapter');
const extensionAdaptor = require('./extensionsSummaryAdapter');
const externalInterestsAdaptor = require('./externalInterestsSummaryAdapter');

const actionSummaryAdapterFactory = {
    appeals: appealAdaptor,
    extensions: extensionAdaptor,
    recordInterest: externalInterestsAdaptor
};

module.exports = {
    actionSummaryAdapterFactory
};