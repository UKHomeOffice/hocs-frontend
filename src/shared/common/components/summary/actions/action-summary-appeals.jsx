import React from 'react';
import PropTypes from 'prop-types';

const renderRow = ({ label, value }) => {
    return (
        <tr key={label} className='govuk-table__row'>
            <th className='govuk-table__header padding-left--small govuk-!-width-one-third'>{label}</th>
            <td className='govuk-table__cell'>{value}</td>
        </tr>
    );
};

renderRow.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
};

const ActionSummaryAppeals = (props) => {

    const appeals = [ ...props.props.items ];

    if (appeals.length < 1) {
        return false;
    }

    return (
        <>
            <h2 className='govuk-heading-m'>Appeals</h2>
            { appeals.map(item => (
                <>
                    <table className='govuk-table margin-left--small'>
                        <caption className='govuk-table__caption margin-bottom--small'>{item.title}</caption>
                        <tbody className='govuk_table__body'>
                            { item.officerDirectorate && renderRow({ label: 'Directorate', value: item.officerDirectorate })}
                            { item.officer && renderRow({ label: 'Officer Name', value: item.officer })}
                            { item.status && renderRow({ label: 'Completed', value: item.status })}
                            { item.dateSentRMS && renderRow({ label: 'Completion date', value: item.dateSentRMS })}
                            { item.outcome && renderRow({ label: 'Outcome', value: item.outcome })}
                            { item.complexCase && renderRow({ label: 'Complex case', value: item.complexCase })}
                            { item.note && renderRow({ label: 'Details', value: item.note })}
                        </tbody>
                    </table>

                </>
            ))}
        </>
    );
};

export default ActionSummaryAppeals;