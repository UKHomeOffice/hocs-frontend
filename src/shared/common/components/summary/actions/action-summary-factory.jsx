import React from 'react';
import AppealSummary from './action-summary-appeals.jsx';
import InterestSummary from './action-summary-interests.jsx';

function renderComponent(Component, action) {
    return (
        <>
            <Component items={action.items} />
        </>
    );
}

const getActionSummary = (actionType, actions) => {
    switch (actionType) {
        case 'Appeals':
            return renderComponent(AppealSummary, actions.find(actions => actions.title === actionType));
        case 'Recordinterest':
            return renderComponent(InterestSummary, actions.find(actions => actions.title === actionType));
        default:
            return false;
    }
};

export default getActionSummary;