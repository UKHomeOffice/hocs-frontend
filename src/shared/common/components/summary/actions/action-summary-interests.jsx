import React from 'react';
import PropTypes from 'prop-types';

const renderRow = ({ label, value }) => {
    return (
        <tr key={label} className='govuk-table__row'>
            <th className='govuk-table__header padding-left--small govuk-!-width-one-third'>{label}</th>
            <td className='govuk-table__cell'>{value}</td>
        </tr>
    );
};

renderRow.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
};

const ActionSummaryInterests = (props) => {

    const interests = [ ...props.props.items ];

    if (interests.length < 1) {
        return false;
    }

    return (
        <>
            <h2 className='govuk-heading-m'>Interest</h2>
            { interests.map((item, index) => (
                <>
                    <table className='govuk-table margin-left--small'>
                        <caption className='govuk-table__caption margin-bottom--small'>Interest { index + 1 }</caption>
                        <tbody className='govuk_table__body'>
                            { item.interestedPartyTitle && renderRow({ label: 'Party', value: item.interestedPartyTitle })}
                            { item.detailsOfInterest && renderRow({ label: 'Details', value: item.detailsOfInterest })}
                        </tbody>
                    </table>

                </>
            ))}
        </>
    );
};

export default ActionSummaryInterests;