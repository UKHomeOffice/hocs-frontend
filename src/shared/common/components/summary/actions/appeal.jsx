import React from 'react';
import PropTypes from 'prop-types';

const renderRow = ({ label, value }) => {
    return (
        <tr key={label} className='govuk-table__row'>
            <th className='govuk-table__header padding-left--small govuk-!-width-one-third'>{label}</th>
            <td className='govuk-table__cell'>{value}</td>
        </tr>
    );
};

renderRow.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
};

const Appeal = ({ appeal }) => {
    return (
        <>
            <table className='govuk-table margin-left--small'>
                <caption className='govuk-table__caption margin-bottom--small'>{appeal.title}</caption>
                <tbody className='govuk_table__body'>
                    { appeal.officerDirectorate && renderRow({ label: 'Directorate', value: appeal.officerDirectorate })}
                    { appeal.officer && renderRow({ label: 'Officer Name', value: appeal.officer })}
                    { appeal.status && renderRow({ label: 'Completed', value: appeal.status })}
                    { appeal.dateSentRMS && renderRow({ label: 'Completion date', value: appeal.dateSentRMS })}
                    { appeal.outcome && renderRow({ label: 'Outcome', value: appeal.outcome })}
                    { appeal.complexCase && renderRow({ label: 'Complex case', value: appeal.complexCase })}
                    { appeal.note && renderRow({ label: 'Details', value: appeal.note })}
                </tbody>
            </table>
        </>
    );
};

Appeal.propTypes = {
    appeal: PropTypes.object.isRequired
};

export default Appeal;
