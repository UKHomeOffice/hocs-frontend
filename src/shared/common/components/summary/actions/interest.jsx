import React from 'react';
import PropTypes from 'prop-types';

const renderRow = ({ label, value }) => {
    return (
        <tr key={label} className='govuk-table__row'>
            <th className='govuk-table__header padding-left--small govuk-!-width-one-third'>{label}</th>
            <td className='govuk-table__cell'>{value}</td>
        </tr>
    );
};

renderRow.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
};

const Interest = ({ interest }) => {
    return (
        <>
            <table className='govuk-table margin-left--small'>
                <caption className='govuk-table__caption margin-bottom--small'>Interest</caption>
                <tbody className='govuk_table__body'>
                    { interest.interestedPartyTitle && renderRow({ label: 'Party', value: interest.interestedPartyTitle })}
                    { interest.detailsOfInterest && renderRow({ label: 'Details', value: interest.detailsOfInterest })}
                </tbody>
            </table>
        </>
    );
};

Interest.propTypes = {
    interest: PropTypes.object.isRequired
};

export default Interest;
