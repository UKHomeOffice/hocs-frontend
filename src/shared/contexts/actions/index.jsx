import types from './types.jsx';

export function updateForm(form) {
    return {
        type: types.UPDATE_FORM,
        payload: form
    };
}

export function updateFormData(data) {
    return {
        type: types.UPDATE_FORM_DATA,
        payload: data
    };
}

export function updateCaseSummary(data) {
    return {
        type: types.UPDATE_CASE_SUMMARY,
        payload: data
    };
}

export function updateCaseConfig(data) {
    return {
        type: types.UPDATE_CASE_CONFIG,
        payload: data
    };
}

export function updateCaseData(data) {
    return {
        type: types.UPDATE_CASE_DATA,
        payload: data
    };
}

export function unsetCaseData() {
    return {
        type: types.UPDATE_CASE_DATA,
        payload: null
    };
}

export function updateCaseActionData(data) {
    return {
        type: types.UPDATE_CASE_ACTION_DATA,
        payload: data
    };
}

export function unSetCaseActionData() {
    return {
        type: types.UNSET_CASE_ACTION_DATA,
        payload: null
    };
}

export function updateFormErrors(data) {
    return {
        type: types.UPDATE_FORM_ERRORS,
        payload: data
    };
}

export function setError(error) {
    return {
        type: types.SET_ERROR,
        payload: error
    };
}

export function unsetError() {
    return {
        type: types.UNSET_ERROR,
    };
}

export function unsetForm() {
    return {
        type: types.UNSET_FORM,
    };
}

export function updateLocation(location) {
    return {
        type: types.UPDATE_LOCATION,
        payload: location
    };
}

export function cancel() {
    return {
        type: types.CANCEL
    };
}

export function redirected() {
    return {
        type: types.REDIRECTED
    };
}

export function redirect(url) {
    return {
        type: types.REDIRECT,
        payload: url
    };
}

export function updateApiStatus(status) {
    return {
        type: types.UPDATE_API_STATUS,
        payload: {
            status,
            timeStamp: Date.now()
        }
    };
}

export function clearApiStatus() {
    return {
        type: types.CLEAR_API_STATUS
    };
}

export function updatePageMeta(meta) {
    return {
        type: types.UPDATE_PAGE_META,
        payload: meta
    };
}

export function updateDashboard(data) {
    return {
        type: types.UPDATE_DASHBOARD,
        payload: data
    };
}

export function clearDashboard() {
    return {
        type: types.CLEAR_DASHBOARD,
        payload: null
    };
}

export function updateWorkstack(data) {
    return {
        type: types.UPDATE_WORKSTACK,
        payload: data
    };
}

export function clearWorkstack() {
    return {
        type: types.CLEAR_WORKSTACK,
        payload: null
    };
}

export function unsetCaseSummary() {
    return {
        type: types.UNSET_CASE_SUMMARY,
        payload: null
    };
}

export function unsetCorrespondents() {
    return {
        type: types.UNSET_CORRESPONDENTS,
        payload: null
    };
}

export function unsetCaseNotes() {
    return {
        type: types.UNSET_CASE_NOTES,
        payload: null
    };
}

export function unsetCaseConfig() {
    return {
        type: types.UNSET_CASE_CONFIG,
        payload: null
    };
}

export function passForwardProps(props) {
    return {
        type: types.PASS_FORWARD_PROPS,
        payload: props
    };
}

export function unsetDocuments() {
    return {
        type: types.UNSET_DOCUMENTS
    };
}
