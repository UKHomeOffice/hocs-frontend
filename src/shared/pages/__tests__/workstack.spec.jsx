import React from 'react';
import WrappedWorkstackPage from '../workstack.jsx';

const BASE_URL = '/api';
const MOCK_MATCH = { url: '/test' };

jest.mock('axios', () => ({
    get: jest.fn((url) => {
        switch (url) {
        case '/api/test':
            return Promise.resolve({
                data: { label: 'Workstack', items: [] }
            });
        default:
            return Promise.reject({
                response: {
                    data: {}
                }
            });
        }
    })
}));
jest.mock('../../common/components/workstack.jsx', () => () => 'MOCK_WORKSTACK');
jest.mock('../../contexts/actions/index.jsx', () => ({
    updateApiStatus: jest.fn(),
    clearApiStatus: jest.fn(),
    updateWorkstack: jest.fn(),
    clearWorkstack: jest.fn(),
}));

import axios from 'axios';
import actions from '../../contexts/actions/index.jsx';
const FLUSH_PROMISES = () => new Promise(resolve => setImmediate(resolve));

describe('Workstack page component', () => {

    const MOCK_DISPATCH = jest.fn();

    beforeEach(() => {
        MOCK_DISPATCH.mockReset();
        MOCK_DISPATCH.mockReturnValue(Promise.resolve());
        axios.get.mockClear();
        actions.updateWorkstack.mockReset();
    });

    it('should render with default props', async () => {
        const OUTER = shallow(<WrappedWorkstackPage match={MOCK_MATCH} />);
        const WorkstackPage = OUTER.props().children;
        const WRAPPER = mount(
            <WorkstackPage dispatch={MOCK_DISPATCH} />
        );
        await FLUSH_PROMISES();
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
        expect(MOCK_DISPATCH).toHaveBeenCalled();
        expect(actions.updateWorkstack).toHaveBeenCalled();
        expect(axios.get).toHaveBeenCalled();
        expect(axios.get).toHaveBeenCalledWith(BASE_URL + MOCK_MATCH.url);
    });

    it('should should not call workstack endpoint if data exists in props', async () => {
        const MOCK_WORKSTACK_DATA = { label: 'Workstack', items: [] };
        const OUTER = shallow(<WrappedWorkstackPage  match={MOCK_MATCH} />);
        const WorkstackPage = OUTER.props().children;
        const WRAPPER = mount(
            <WorkstackPage dispatch={MOCK_DISPATCH} workstack={MOCK_WORKSTACK_DATA} />
        );
        await FLUSH_PROMISES();
        expect(WRAPPER).toBeDefined();
        expect(MOCK_DISPATCH).not.toHaveBeenCalled();
        expect(actions.updateWorkstack).not.toHaveBeenCalled();
        expect(axios.get).not.toHaveBeenCalled();
    });

});