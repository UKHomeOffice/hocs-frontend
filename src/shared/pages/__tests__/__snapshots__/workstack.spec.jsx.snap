// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workstack page component should render with default props 1`] = `
<Component
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
        Array [
          undefined,
        ],
        Array [
          undefined,
        ],
      ],
    }
  }
>
  <WorkstackPage
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
          Array [
            undefined,
          ],
          Array [
            undefined,
          ],
        ],
      }
    }
    match={
      Object {
        "url": "/test",
      }
    }
  >
    <p
      className="govuk-body"
    >
      No items to display
    </p>
  </WorkstackPage>
</Component>
`;
